runtime/src/main/cpp/Memory.cpp

```diff
@@ -1752,11 +1751,13 @@ void deinitMemory(MemoryState* memoryState) {
   atomicAdd(&pendingDeinit, 1);
 #if USE_GC
   bool lastMemoryState = atomicAdd(&aliveMemoryStatesCount, -1) == 0;
-  bool checkLeaks = g_checkLeaks && lastMemoryState;
+  bool checkLeaks = Kotlin_memoryLeakCheckerEnabled() && lastMemoryState;
   if (lastMemoryState) {
    garbageCollect(memoryState, true);
 #if USE_CYCLIC_GC
    // If there are other pending deinits (rare situation) - just skip the leak checker.
+   // This may happen when there're several threads with Kotlin runtimes created
+   // by foreign code, and that code stops those threads simultaneously.
```

Comment:
Is this still true?

Reply:
Yes. Memory leak checker still runs on "foreign" threads termination, it's not bound to `main()` termination.